presto-rcfile/src/main/java/com/facebook/presto/rcfile/binary/BooleanEncoding.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.rcfile.binary;
+
+import com.facebook.presto.rcfile.ColumnData;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.Slice;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+
+public class BooleanEncoding
+        implements BinaryColumnEncoding
+{
+    @Override
+    public Block decodeColumn(Type type, ColumnData columnData)
+    {
+        int size = columnData.rowCount();
+        BlockBuilder builder = type.createBlockBuilder(new BlockBuilderStatus(), size);
+
+        Slice slice = columnData.getSlice();
+        for (int i = 0; i < size; i++) {
+            int length = columnData.getLength(i);
+            if (length != 0) {
+                type.writeBoolean(builder, slice.getByte(columnData.getOffset(i)) != 0);
```

Comment:
Add `checkState(length == SIZE_OF_BYTE, "Bytes should be 1 byte")`, just like you did with other encodings.